docket/plugins/config_scripts.py

```diff
@@ -0,0 +1,167 @@
+from docket.plugins._group import docket_group
+
+plugin = crescent.Plugin(
+    "config-scripts", command_hooks=[has_guild_perms(hikari.Permissions.MANAGE_GUILD)]
+)
+scripts = docket_group.sub_group("scripts", "Manage scripts")
+
+
+async def wait_for_script(ctx: crescent.Context) -> str | None:
+    def predicate(message: hikari.GuildMessageCreateEvent) -> bool:
+        return (
+            message.channel_id == ctx.channel_id
+            and message.author.id == ctx.user.id
+            and (message.content is not None or message.message.attachments is not None)
+        )
+
+    try:
+        msg = await ctx.app.wait_for(
+            hikari.GuildMessageCreateEvent, timeout=30, predicate=predicate
+        )
+    except asyncio.TimeoutError:
+        return None
+
+    if not msg.message.content:
+        attachment = msg.message.attachments[0]
+        content = str(await attachment.read())
+    else:
+        content = msg.message.content
+        content = msg.message.content.strip("```lua\n").strip("```")
+    return content
+
+
+async def script_name_autocomplete(
+    ctx: crescent.Context, option: hikari.AutocompleteInteractionOption
+) -> list[hikari.CommandChoice]:
+    if not ctx.guild_id:
+        return []
+    prefix = cast(str, option.value)
+    scripts = await Script.fetch_query().where(guild_id=ctx.guild_id).fetchmany()
+    script_names = [script.name for script in scripts]
+    return [
+        hikari.CommandChoice(name=name, value=name)
+        for name in get_close_matches(prefix, script_names, 10, 0.2)
+    ]
+
+
+@plugin.include
+@scripts.child
+@crescent.command(name="create", description="Create a new script")
+class CreateScript:
+    name = crescent.option(str, "The name of the script")
+
+    async def callback(self, ctx: crescent.Context) -> None:
+        name = self.name.lower().casefold().replace(" ", "-")
+        assert ctx.guild_id
+        await ctx.respond(
+            "Please send the Lua script (either in a code block or file " "upload)."
+        )
+        content = await wait_for_script(ctx)
+        if not content:
+            await ctx.edit("Script creation timed out.")
+
+        await Guild.get_or_create(ctx.guild_id)
+        try:
+            await Script(guild_id=ctx.guild_id, name=name, code=content).create()
+        except asyncpg.UniqueViolationError:
+            await ctx.edit("A script with that name already exists.")
+            return
+
+        await ctx.edit(f"Script '{name}' created.")
+
+
+@plugin.include
+@scripts.child
+@crescent.command(name="delete", description="Delete a script")
+class DeleteScript:
+    name = crescent.option(
+        str, "The name of the script", autocomplete=script_name_autocomplete
+    )
+
+    async def callback(self, ctx: crescent.Context) -> None:
+        assert ctx.guild_id
+        script = await Script.exists(name=self.name, guild_id=ctx.guild_id)
+        if not script:
+            raise ScriptNotFound(self.name)
```

Comment:
You reuse this block of code. I think it can be a function.

Reply:
Maybe? Seems a little trivial tbh.

Reply:
`ScriptNotFound` is closely related to the `Script` object i think it makes much more sense for it to be raised from one of its methods